django/core/mail/backends/locmem.py

```diff
@@ -26,6 +27,6 @@ def send_messages(self, messages):
         msg_count = 0
         for message in messages:  # .message() triggers header validation
             message.message()
-            mail.outbox.append(message)
+            mail.outbox.append(copy.deepcopy(message))
```

Comment:
Is it worth calling `deepcopy()`?

Reply:
Deep copying is worth it here since a shallow copy wouldn't protect against mutations to the `to` list. Example:
```py
def test_mutate_after_send(self):
        email = EmailMessage(
            subject="correct subject",
            body="test body",
            from_email="from@example.com",
            to=["to@example.com"],
        )
        email.send()
        email.to.append("incorrect@example.com")
        self.assertEqual(["to@example.com"], mail.outbox[0].to)
```
This test fails when using a shallow copy but succeeds with a deep copy

Reply:
I have now updated the test to account for this 👍